apis/rbac/v1alpha1/types.go

```diff
@@ -0,0 +1,302 @@
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories={crossplane,managed,azure}
+type Application struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ApplicationSpec   `json:"spec"`
+	Status ApplicationStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// ApplicationList contains a list of Application items
+type ApplicationList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Application `json:"items"`
+}
+
+// An ServicePrincipalSpec defines the desired state of an ServicePrincipal.
+type ServicePrincipalSpec struct {
+	xpv1.ResourceSpec `json:",inline"`
+
+	// ApplicationID - The application ID.
+	// +optional
+	ApplicationID string `json:"applicationID,omitempty"`
+
+	// ApplicationIDRef - A reference to the Application id.
+	// +optional
+	// +immutable
+	ApplicationIDRef *xpv1.Reference `json:"applicationIDRef,omitempty"`
+
+	// ApplicationIDSelector - Select a reference to the Application id.
+	// +immutable
+	ApplicationIDSelector *xpv1.Selector `json:"applicationIDSelector,omitempty"`
+
+	// AccountEnabled - whether or not the service principal account is enabled
+	// +optional
+	// +immutable
+	AccountEnabled *bool `json:"accountEnabled,omitempty"`
+}
+
+// An ServicePrincipalStatus represents the observed state of an ServicePrincipal.
+type ServicePrincipalStatus struct {
+	xpv1.ResourceStatus `json:",inline"`
+}
+
+// +kubebuilder:object:root=true
+
+// A ServicePrincipal is a managed resource that represents an ServicePrincipal.
+// +kubebuilder:printcolumn:name="READY",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="SYNCED",type="string",JSONPath=".status.conditions[?(@.type=='Synced')].status"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories={crossplane,managed,azure}
+type ServicePrincipal struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ServicePrincipalSpec   `json:"spec"`
+	Status ServicePrincipalStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// ServicePrincipalList contains a list of ServicePrincipal items
+type ServicePrincipalList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ServicePrincipal `json:"items"`
+}
+
+// An RoleAssignmentSpec defines the desired state of an RoleAssignment.
+type RoleAssignmentSpec struct {
+	xpv1.ResourceSpec `json:",inline"`
+
+	// PrincipalType - type of principal oneof: ['group', 'service-principal']
```

Comment:
Could you please introduce a `RoleAssignmentParameters` struct containing these fields? Also per Crossplane [API conventions](https://github.com/crossplane/crossplane/blob/master/design/one-pager-managed-resource-api-design.md), this field should be under `spec.forProvider` and not inlined.  For an example, please see:
https://github.com/crossplane/provider-azure/blob/1c0702110ed45feabe2af15a0478e753d1c34d4e/apis/database/v1beta1/configuration_types.go#L103